app.js

```diff
@@ -11,72 +11,89 @@ const teamTwoSubtractButton = document.getElementById('team-two-subtract-button'
 const teamOneLabel = document.getElementById('team-one-name');
 const teamTwoLabel = document.getElementById('team-two-name');
+const teamOneInput = document.getElementById('team-one-input');
+const teamTwoInput = document.getElementById('team-two-input');
 
 // create an array to hold on to the state of past games
+const pastGames = [];
 
 let name1 = '';
-let name2 =  '';
+let name2 = '';
 let score1 = 0;
 let score2 = 0;
 
 nameFormButton.addEventListener('click', () => {
     // get the name data from the form
-
+    const teamOne = teamOneInput.value;
+    const teamTwo = teamTwoInput.value;
     // set the state to this data from the form
-
+    name1 = teamOne;
+    name2 = teamTwo;
     // reset the form values
-
+   
     // refresh the current game element with new data by calling the appropriate function
+    displayCurrentGameEl();
 });
 
 
 teamOneAddButton.addEventListener('click', () => {
-    // increment the current state for team one's score
-    
-    // refresh the current game element with new data by calling the appropriate function
+   
+    score1++;
+
+    displayCurrentGameEl();
 });
 
 teamTwoAddButton.addEventListener('click', () => {
-    // increment the current state for team two's score
-
-    // refresh the current game element with new data by calling the appropriate function
+  
+    score2++;
+  
+    displayCurrentGameEl();
 });
 
 teamOneSubtractButton.addEventListener('click', () => {
-    // decrement the current state for team one's score
-
-    // refresh the current game element with new data by calling the appropriate function
+  
+    score1--;
+   
+    displayCurrentGameEl();
 });
 
 teamTwoSubtractButton.addEventListener('click', () => {
-    // decrement the current state for team two's score
-
-    // refresh the current game element with new data by calling the appropriate function
+  
+    score2--;
+  
+    displayCurrentGameEl();
 });
 
 finishGameButton.addEventListener('click', () => {
     
-    // add the current game to an array of games in state
-    // it will be helpful to keep track of these games as objects with 4 properties, one for each piece of state we're tracking
-    // for example, make an object like this: { name1: 'ducks', name2: 'bears' ,score1: 1, score2: 2 } 
-    // then push it to your array in state
-    // (be sure to make a new object. do not declare the object in global scope and mutate it for reuse. This would cause difficult bugs)
-    
+    const currentGameData = {
+        name1: name1,
+        name2: name2,
+        score1: score1,
+        score2: score2,
+    };
+    pastGames.push(currentGameData);
+    pastGamesEl.textContent = '';
+
+   
     displayAllGames();
-
-    // reset the state to zero and empty strings
-    
-    // refresh the current game element with new data by calling the appropriate function
+    name1 = '';
+    name2 = '';
+    score1 = 0;
+    score2 = 0;
+    currentGameEl.textContent = '';
```

Comment:
This line isn't necessary here since it's the first thing that happens in displayCurrrentGameEl() which is on the next line.